src/org/opendatakit/briefcase/ui/reused/UiFieldValidator.java

```diff
@@ -0,0 +1,74 @@
+package org.opendatakit.briefcase.ui.reused;
+
+import java.awt.Color;
+import java.awt.event.KeyAdapter;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+import javax.swing.JLabel;
+import javax.swing.JTextField;
+import org.opendatakit.briefcase.reused.http.RequestBuilder;
+
+public class UiFieldValidator {
+  public static final Predicate<String> REQUIRED = s -> s != null && !s.trim().isEmpty();
+  public static final Predicate<String> NUMBER = s -> s == null || s.matches("\\d+");
+  public static final Predicate<String> EMAIL = s -> s == null || s.matches("\\S+@\\S+");
+  public static final Predicate<String> URI = s -> s == null || RequestBuilder.isUri(s);
+  static final Color COLOR_NOT_VALID = new Color(215, 25, 28); // Color-blind safe red
+  static final Color COLOR_VALID = new Color(44, 123, 182); // Color-blind safe blue
```

Comment:
Very thoughtful with the color-blind-safe colors

Reply:
It was Yaw's idea :)